src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -88,48 +109,37 @@ public AvroSerializer(ISchemaRegistryClient schemaRegistryClient)
         { 
-            if (serializerImpl == null)
-            {
-                serializerImpl = typeof(T) == typeof(GenericRecord)
-                    ? (IAvroSerializerImpl<T>)new GenericSerializerImpl(schemaRegistryClient, autoRegisterSchema, initialBufferSize, isKeySerializer)
-                    : new SpecificSerializerImpl<T>(schemaRegistryClient, autoRegisterSchema, initialBufferSize, isKeySerializer);
-            }
-
-            return serializerImpl.Serialize(topic, data);
-        }
-
-        /// <summary>
-        ///     Configure the serializer.
-        /// </summary>
-        public IEnumerable<KeyValuePair<string, string>> Configure(IEnumerable<KeyValuePair<string, string>> config, bool isKey)
-        {
-            var keyOrValue = isKey ? "Key" : "Value";
-            var avroConfig = config.Where(item => item.Key.StartsWith("avro."));
-            this.isKeySerializer = isKey;
-
-            if (avroConfig.Count() != 0)
+            try
             {
-                int? initialBufferSize = (int?)Utils.ExtractPropertyValue(config, isKey, AvroSerdeProviderConfig.PropertyNames.AvroSerializerBufferBytes, "AvroSerializer", typeof(int));
-                if (initialBufferSize != null) { this.initialBufferSize = initialBufferSize.Value; }
-
-                bool? autoRegisterSchema = (bool?)Utils.ExtractPropertyValue(config, isKey, AvroSerdeProviderConfig.PropertyNames.AvroSerializerAutoRegisterSchemas, "AvroSerializer", typeof(bool));
-                if (autoRegisterSchema != null) { this.autoRegisterSchema = autoRegisterSchema.Value; }
-
-                foreach (var property in avroConfig)
+                if (serializerImpl == null)
                 {
-                    if (property.Key != AvroSerdeProviderConfig.PropertyNames.AvroSerializerAutoRegisterSchemas && property.Key != AvroSerdeProviderConfig.PropertyNames.AvroSerializerBufferBytes)
-                    {
-                        throw new ArgumentException($"{keyOrValue} AvroSerializer: unexpected configuration parameter {property.Key}");
-                    }
+                    serializerImpl = typeof(T) == typeof(GenericRecord)
+                        ? (IAvroSerializerImpl<T>)new GenericSerializerImpl(schemaRegistryClient, autoRegisterSchema, initialBufferSize)
+                        : new SpecificSerializerImpl<T>(schemaRegistryClient, autoRegisterSchema, initialBufferSize);
                 }
-            }
 
-            return config.Where(item => !item.Key.StartsWith("avro."));
+                return await serializerImpl.Serialize(topic, data, isKey);
+            }
+            catch (AggregateException e)
+            {
+                throw new SerializationException("Error occured serializing avro data.", e.InnerException);
+            }
+            catch (Exception e)
+            {
+                throw new SerializationException("Error occured serializing avro data.", e);
+            }
         }
```

Comment:
`SerializeException`, `SerializerException` or `SerializationException`?

Reply:
ha, well in my current local branch I have `SerializeException`, @rnpridgeon reckons `SerializerException` and @edenhill reckons `SerializationException`. Today my opinion coincides with @edenhill. I'll pick one, probably that.